src/stream/src/task/stream_manager.rs

```diff
@@ -846,6 +865,42 @@ impl LocalStreamManagerCore {
     }
 }
 
+pub async fn allocation_stat_for_stream<Fut, T, F>(
```

Comment:
Copied from `allocation_stat` but with `ctx.store_mem_usage(0)`? What's this for? It looks ugly.

Reply:
Yes, I have to say it's a little ugly. But beasue we need to report 0 bytes when actor drops, so we need to pass another context ref (the report `F` has been moved so I can not call it to report 0 bytes). It seems that do not have better ideas.

Reply:
I doubt if we should report a 0 bytes. Allocated bytes stats include all memory allocated in the actor including those in block cache. When a actor gets dropped, those memory won't be free.

Reply:
Good point. So it's possible that the `BYTES_ALLOCATED` is not 0 when the actor drops, but the actor must be dropped when `BYTES_ALLOCATED` is 0 (Cuz in this case `BYTES_ALLOCATED` will be dropped). cc @TennyZhuang to ensure I'm right.

But seems that `BYTES_ALLOCATED` will only be reproted when our actor is running, so when I drop the actor, even if there is allocated memory in block cache, it will no longer report usage (The report function `allocation_stat` is stop working when actor stops). Anyway I'll try to remove this.

Reply:
Have fix this: do not report 0 bytes, but report the `BYTES_ALLOCATED` value. This looks more reasonable. But still we got two functions: `allocation_stat` and `allocation_stat_for_stream`...